pim-tool-back/src/main/java/vn/elca/training/web/ProjectsController.java

```diff
@@ -0,0 +1,37 @@
+package vn.elca.training.web;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import vn.elca.training.model.dto.ProjectDto;
+import vn.elca.training.service.ProjectService;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * @author gtn
+ * @author thomas.dang
+ */
+@RestController
+@RequestMapping("/projects")
+public class ProjectsController extends AbstractApplicationController {
+
+    @Autowired
+    private ProjectService projectService;
+
+    @GetMapping("/search")
+    public List<ProjectDto> search(@RequestParam String keyword) {
+        if (keyword.isBlank()) {
```

Comment:
[S] Use Apache utility method StringUtils.isEmpty() or isBlank() in order to avoid NPE